src/taskgraph/util/keyed_by.py

```diff
@@ -2,8 +2,40 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+from typing import Any, Dict, Generator, Tuple
 
-from .attributes import keymatch
+from taskgraph.util.attributes import keymatch
+
+
+def iter_dot_path(
+    container: Dict[str, Any], subfield: str
+) -> Generator[Tuple[dict[str, Any], str], None, None]:
```

Comment:
Python < 3.9 can't handle type annotation using built-in types 😢 Just need to replace `dict` with `Dict` (I think it can handle `str` tho). Very annoying.